packages/next/src/lib/worker.ts

```diff
@@ -24,11 +24,36 @@ export class Worker {
     this._worker = undefined
 
     const createWorker = () => {
-      this._worker = new JestWorker(workerPath, farmOptions) as JestWorker
+      this._worker = new JestWorker(workerPath, {
+        ...farmOptions,
+        forkOptions: {
+          ...farmOptions.forkOptions,
+          env: {
+            ...((farmOptions.forkOptions?.env || {}) as any),
+            ...process.env,
+            // we don't pass down NODE_OPTIONS as it can
+            // extra memory usage
+            NODE_OPTIONS: '',
```

Comment:
Does this mean that we're going to be stuck w/ node defaults? Is that enough for the child processes?

Reply:
It should be plenty as we span a worker for each CPU by default and the workers are supposed to retry as well.

Reply:
more info on this from @ijjk:

We have a site that generates 208,630 pages, and we've set the env var  `--max_old_space_size=12288` on the main call to `next build` because its running on a kubernetes pod limited to 16GB memory (we need some overhead for the parent process).

We just upgraded to `v13.2.4` which includes this PR, we wanted to see what effect this release had on memory usage, and now we're seeing a _slightly_ different but interesting behavior in the workers: 

```
// snipped of next build output
info  - Compiled successfully
info  - Collecting page data...
Static worker unexpectedly exited with code: null and signal: SIGKILL
Static worker unexpectedly exited with code: null and signal: SIGKILL
Static worker unexpectedly exited with code: null and signal: SIGTERM
warn  - Restarted collecting page data for [object Object] because it took more than 600 seconds
warn  - See more info here https://nextjs.org/docs/messages/static-page-generation-timeout
warn  - Restarted collecting page data for [object Object] because it took more than 600 seconds
info  - Generating static pages (0/208630)
Static worker unexpectedly exited with code: null and signal: SIGKILL
Static worker unexpectedly exited with code: null and signal: SIGKILL
Static worker unexpectedly exited with code: null and signal: SIGKILL
info  - Generating static pages (732/208630)
info  - Generating static pages (3750/208630)
// snipped
info  - Generating static pages (208630/208630)
info  - Finalizing page optimization...
success
```
but despite these errors the build ended up completing successfully.

| ![Screen Shot 2023-04-01 at 9 47 49 AM](https://user-images.githubusercontent.com/51207/229293981-742f1673-a8c6-4c97-8615-ae2215a52768.png) |
|:---:|
| memory usage during `next build` |

Reply:
| ![Screen Shot 2023-04-01 at 10 06 37 AM](https://user-images.githubusercontent.com/51207/229294141-7825d7a7-8a31-4fc9-a8e5-8b6913bfeea0.png) |
|:---:|
| _memory initially spikes, service workers are killed, and memory usage gradually increases until the process finishes_ |